e2e/test/scenarios/models/reproductions/31309-duplicating-a-model-must-duplicate-question.cy.spec.js

```diff
@@ -0,0 +1,100 @@
+import {
+  getNotebookStep,
+  modal,
+  openQuestionActions,
+  popover,
+  restore,
+} from "e2e/support/helpers";
+
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import { SAMPLE_DB_ID } from "e2e/support/cypress_data";
+
+const { ORDERS, ORDERS_ID, PEOPLE } = SAMPLE_DATABASE;
+
+const TEST_QUERY = {
+  "order-by": [["asc", ["field", "sum", { "base-type": "type/Float" }]]],
+  limit: 10,
+  filter: ["<", ["field", "sum", { "base-type": "type/Float" }], 100],
+  "source-query": {
+    "source-table": ORDERS_ID,
+    aggregation: [["sum", ["field", ORDERS.TOTAL, null]]],
+    breakout: [
+      [
+        "field",
+        PEOPLE.NAME,
+        { "base-type": "type/Text", "source-field": ORDERS.USER_ID },
+      ],
+    ],
+  },
+};
+
+describe("scenarios > models with aggregation and breakout", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+    cy.intercept("POST", "/api/card").as("createModel");
+  });
+
+  it("should duplicate a model with its original aggregation and breakout", () => {
+    cy.createQuestion(
+      {
+        name: "model",
+        query: TEST_QUERY,
+        type: "query",
+        database: SAMPLE_DB_ID,
+        dataset: true,
+      },
+      {
+        visitQuestion: true,
+      },
+    );
+
+    openQuestionActions();
+    popover().within(() => {
+      cy.findByText("Duplicate").click();
+    });
+
+    modal().within(() => {
+      cy.findByText("Duplicate").click();
+    });
+
+    modal().within(() => {
+      cy.findByText("Not now").click();
+    });
+
+    cy.wait("@createModel").then(({ response: { body } }) => {
+      cy.request("GET", `/api/card/${body.id}`).then(
+        ({ body: { dataset_query } }) => {
+          expect(dataset_query.query).to.deep.eq(TEST_QUERY);
+          expect(dataset_query.database).to.eq(SAMPLE_DB_ID);
+          expect(dataset_query.type).to.eq("query");
+        },
+      );
+    });
+
+    openQuestionActions();
```

Comment:
These tests after the `GET` request test make sure the UI elements are presented in the model query definition editor. Obviously, we've already checked that the model has the same query within the database, so this is more of a double-check than anything else. We can remove this if anybody thinks this is a little too redundant.

Reply:
> We can remove this if anybody thinks this is a little too redundant.

I'm interested to see what others think.

----

My 2 cents: after reading the title of this test case I didn't expect to find these checks here.

Reply:
Gotcha, I see what you mean. Do you think these tests still provide any value (and modify the test name to make it more clear) or should we nuke them?

Reply:
> I'd remove the API response check and leave the UI check here.

Why remove them?
I'd keep both API and UI checks.
This way, if something breaks, we can narrow it down to where the bug is (before or after the API call) quicker.

Reply:
In general, we should test user interactions and app behaviors. Asserting API is testing against the implementation. One point of an End-to-End test is to make us more confident that the feature won't break when we modify the code around that area e.g. refactoring, fixing bugs, improving or extending features. And asserting and API could break that assumption. The app behavior could be the same but has some API changes which could cause the test to fail.

> I'd keep both API and UI checks.
This way, if something breaks, we can narrow it down to where the bug is (before or after the API call) quicker.

This is a good point, but that assumes the API is rigid which won't change. Asserting API might be a good idea if we want to preserve API contracts e.g. if we have API versioning, but Metabase never does API versioning or promises that we will never change the API, so it's safe to say that API can be changed when needed.